### PR TITLE
Add courses page

### PR DESCRIPTION
**Pull request by ygorgs**
Create the courses page using the table component.

image

### CODE REVIEW

File: src/actions/courseAction.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  return dispatch => {
+    fetchGet('/api/course/')
+      .then(courses => {
+        dispatch({ type: action.GET_COURSES, courses: courses });
+      })
+      .catch(error => {
+        console.log(error)
+      })
+  }
+};
```
Comment (CharlleDaniel) [inline src/actions/courseAction.js]:
```suggestion
};

```

File: src/containers/courses/courses.js [hunk 2]
```diff
@@ -1,11 +1,52 @@
...
+    date: 'Created at',
+    enrolled: 'Enrolled',
+    actions: 'Actions' // check how we're going to deal with solving
+  };
+
+  componentDidMount() {
+    this.props.onInitCourses()
+  }
+
+  renderContent() {
```
Comment (CharlleDaniel) [inline src/containers/courses/courses.js]:
```suggestion
  renderContent() {
```

Comment (CharlleDaniel) [reply src/containers/courses/courses.js]: Please move all code inside this method to `render()`.

File: src/reducers/coursesReducer.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const CoursesReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case actionTypes.GET_COURSES:
+      return {
+        error: null,
+        isLoading: false,
+        cources: action.courses
+      };
+    default: return state
+
```
Comment (CharlleDaniel) [inline src/reducers/coursesReducer.js]:
```suggestion

```

Comment (CharlleDaniel) [reply src/reducers/coursesReducer.js]: please, remove this line.

File: src/containers/courses/courses.js [hunk 4]
```diff
@@ -1,11 +1,52 @@
 import React, { Component } from 'react';
+import {connect} from "react-redux";
+import Table from '../../components/utils/table'
+import * as courseAction from "../../actions/courseAction";
+import './style/courses.css';
+
+class Courses extends Component {
+
+  headers = {
```
Comment (CharlleDaniel) [inline src/containers/courses/courses.js]: Please, define `headers` as `const HEADERS_TABLE` outside this class.

File: src/actions/course-action.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import * as action from './actions';
+import { fetchGet } from '../shared/utility'
+
+export const getCourses = () => {
+  return dispatch => {
+    fetchGet('/api/course/')
```
Comment (CharlleDaniel) [inline src/actions/course-action.js]:
```suggestion
    fetchGet('/course/')
```

File: src/actions/actions.js [hunk 6]
```diff
@@ -1,7 +1,9 @@
...
-export const DELETE_PROBLEM = "DELETE_PROBLEM";
-export const UPDATE_PROBLEM = "UPDATE_PROBLEM";
-export const CREATE_FAIL = "CREATE_FAIL";
+export const INIT_PROBLEMS = 'INIT_PROBLEMS';
+export const CREATE_PROBLEM = 'CREATE_PROBLEM';
+export const DELETE_PROBLEM = 'DELETE_PROBLEM';
+export const UPDATE_PROBLEM = 'UPDATE_PROBLEM';
+export const CREATE_FAIL = 'CREATE_FAIL';
+
+export const GET_COURSES = 'GET_COURSES';
```
Comment (CharlleDaniel) [inline src/actions/actions.js]:
```suggestion
export const GET_COURSES = "GET_COURSES";
```

File: src/components/ide/ide.js [hunk 7]
```diff
@@ -8,4 +8,4 @@ export default class Ide extends Component {
       </div>
     )
   }
-}
+}
```
Comment (CharlleDaniel) [inline src/components/ide/ide.js]: @ygorgs  this modification isn't necessary on this PR. please remove it.

File: src/actions/actions.js (hunk 6 above)
Comment (ygorgs) [reply src/actions/actions.js]: Single quotes is the pattern in web development

Comment (CharlleDaniel) [reply src/actions/actions.js]: @ygorgs  We are using duble quotes as pattern.

File: src/components/menu/menu.css [hunk 8]
```diff
@@ -1,42 +1,57 @@
-.navbar-default .navbar-nav > .active > span {
```
Comment (CharlleDaniel) [inline src/components/menu/menu.css]: This file is reverting the fix already merged on the menu. Please remove this modification.

**Pull request closed without merging**